Fix access of mediaSrc value

jhonatangcavalcanti opened the pull request:
### This PR will...

Fix the access to retrieve the `src` value from current media.

The problem happens when there is a video element child which is not a `source` element. Other children could be `track` or any other element.

Now, instead of retrieving the `src` from `firstChild`, it is retrieved from the `source` element if it exists.

### Why is this Pull Request needed?

The cleanup when detaching media is being skipped and an error is wrongly logged due to this problem.

<img width="753" alt="Screenshot 2024-08-03 at 16 29 56" src="https://github.com/user-attachments/assets/51e04c37-3618-4716-90b4-e621fca6bd63">


### Are there any points in the code the reviewer needs to double check?

### Resolves issues:

### Checklist

- [X] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] API or design changes are documented in API.md

jhonatangcavalcanti commented:
Hello, @robwalch. Is it possible to add this patch to a new version `1.5.4` on top of `1.5.3`?

robwalch commented:
> Hello, @robwalch. Is it possible to add this patch to a new version `1.5.4` on top of `1.5.3`?

Please open a PR against the "patch/v1.5.x" branch.

The latest release is v1.5.13 so that would get merged and released at v1.5.14.

Consider that there may be integrations or platforms where querySelector is not a method available on the object attached (not an actual HTMLMediaElement but an object with a src property).

jhonatangcavalcanti commented:
Thanks, @robwalch. I did update this PR pointing to `patch/v1.5.x`.
I've also added the ac1c6668306c836a700ccb3a64abbaec8408f230 to keep compatibility with the current implementation of getting the mediaSrc from firstChild.

Is it ok?

jhonatangcavalcanti commented:
I'll need to update it again, the new implementation brings back the problem that I've reported.
I'll work on a new commit with a fix

jhonatangcavalcanti commented:
I updated and added a check to ensure that `media.querySelector` is available before executing it.

I don't know if there is a scenario where the `media` is not a `video` tag and the `firstChild` is a valid element with a `src` attribute.

robwalch commented on src/controller/buffer-controller.ts:
```diff
@@ -1026,9 +1026,9 @@ export default class BufferController implements ComponentAPI {
-    const media =
-      (this.media?.firstChild as HTMLSourceElement | null) || this.media;
-    return media?.src;
+    if (!this.media) return undefined;
+    const media = getSourceChild(this.media) || this.media;
+    return media.src;
```
Can we keep this closer to the original and drop `getSourceChild`?

```suggestion
    const media = this.media?.querySelector?.('source') || this.media;
    return media?.src;
```

jhonatangcavalcanti replied:
Yes, done on 13301b49cd4cdd3979b25dd3b232007fcbe12680

robwalch commented:
@jhonatangcavalcanti,

Thank you for the fix. This has been released in v1.5.14 and merged into dev.
